app/src/main/java/com/eveningoutpost/dexdrip/utilitymodels/BroadcastGlucose.java

```diff
@@ -147,7 +151,13 @@ public static void sendLocalBroadcast(final BgReading bgReading) {
                 }
 
                 bundle.putInt(Intents.EXTRA_SENSOR_BATTERY, BridgeBattery.getBestBridgeBattery());
-                bundle.putLong(Intents.EXTRA_SENSOR_STARTED_AT, sensor.started_at);
+                if (getBestCollectorHardwareName().equals("G7")) {// If we are using G7 or One+
+                    if (FirmwareCapability.isDeviceG7(getTransmitterID())) { // Only if there is connectivity
+                        bundle.putLong(Intents.EXTRA_SENSOR_STARTED_AT, DexSessionKeeper.getStart());
```

Comment:
I think you should read  `DexSessionKeeper.getStart()` in to a local variable and check it is > 0 as part of the overall logic because otherwise you are assuming it is valid. Also shouldn't all these conditions be part of a single condition and if not met then use the existing behavior. Here it looks like you would omit the value if there isn't data. Is that what you want or do you want to emit the sensor start time in this case?

Reply:
We have another problem, which is the sensor start time is not valid for G7.  You were not happy with my approach that addressed both issues here: https://github.com/NightscoutFoundation/xDrip/pull/3263

This PR is only addressing the broadcast for G7 without addressing the incorrect sensor start time.

When someone is using G7 and they change the transmitter ID in order to switch to their new G7, before connectivity is established, the sensor start time should not be broadcasted because it is incorrect.
That's why if we are using G7, but there is no connectivity, which is the case between two sensors for a few read cycles, this PR will not make any broadcast.

Reply:
I don't understand your first statement.
If we are using G7 and if we know what the firmware is, how could Dex start time be incorrect considering that is what we show on the system status page.

Reply:
It is fine if you don't think we should broadcast anything about the session start time for g7 until we are sure. That is why I asked if it was intended.

Your code assumes that `getStart()` will have valid data. I believe that as things are at the moment this will be the case if we have data in `FirmwareCapability` but I don't believe they are actually linked and if something changed in the future, for example not reading the session start time right away (which is unlikely but possible as we might have to read firmware version first) then this new code will break.

That is why I am suggesting that you also check that `getStart()` returns >0 as part of your condition before broadcasting data because the whole purpose of this is not to broadcast invalid data. If the session keeper is not initialized yet then you will get 0 as the default value I believe. Best to check >0 in case I ever make -1 the default as it is for some other things.

Reply:
Understood.  Makes perfect sense.  Thanks.  Please let me fix it.

Reply:
Done